cilium/CFP-39876-clustermesh-filtered-export.md

```diff
@@ -0,0 +1,156 @@
+ 
+## Network Policy Support 
+# Cross-Cluster Communication and Network Policy Behavior
+
+Full support for **cross-cluster communication** and **network policy enforcement** is available **only when both the source (client) and destination (server) pods**—which may optionally be accessed via a service—**are part of a _global namespace_**.
+
+### Behavior When Pods Are Not in Global Namespaces
+
+If either the source or the destination pod does **not** belong to a global namespace, cross-cluster communication **may still succeed**, but under certain conditions although we do not support as part of the new mode. The evaluation of network policy for different cases is summarize below:
+
+
+### Summary of Enforcement Logic
+
+| Source Pod in Global NS | Destination Pod in Global NS | Network Policy Applies | Traffic Allowed? |
+|-------------------------|-------------------------------|-------------------------|------------------|
+| Yes                     | Yes                           | Yes                     | ✅ Enforced normally |
+| No                      | Yes                           | Ingress Policy          | ❌ Works for tunnel mode. Does not work for native routing mode  |
+| Yes                     | No                            | Egress Policy           | ❌ Works for tunnel mode. Does not work for native routing mode |
+| No                      | No                            | Any Policy              | ❌ Works for tunnel mode. Does not work for native routing mode  |
+| Any                     | Any                           | No Policies             | ✅ Allowed (no restrictions) |
+
+Even though the policies work for tunnel routing mode, for consistency in customer experience, we do not provide support for network policies for workloads outside global namespaces. 
+
+## Global Service Support
+In the scoped-export mode, all services within global namespaces would automatically be marked as "global". Customers are not required to explicitly annotate individual services. Also, a service which is annotated as global but does not reside in a global namespaces would no longer be global.  
+
+
+### Implementation details
+
+#### Allow/Deny listing namespace
+To allowlist a namespace and mark it as global, customers can annotate the namespace with:
+
+```
+namespace.cilium.io/global: "true"
+```
+
+The ClusterMesh API server will watch namespace resources. Adding this annotation to any namespace will result in the following steps:
+
+- A full synchronization is triggered 
+- Only entries associated with namespaces explicitly annotated as global will be exported.
+- Entries in etcd that do not belong to any global (allowlisted) namespace will be deleted.
+- All unannotated namespaces or entries with denied annotations are implicitly treated as non-global and excluded from export.
+
+To explicitly denylist a namespace, customers can annotate it with:
+
+```
+namespace.cilium.io/local: "true"
+```
```

Comment:
Could you explain why the two annotation instead of one? For instance in what way `namespace.cilium.io/global: "false"` is different from `namespace.cilium.io/local: "true"`?

Reply:
They are different due to handling of unannotated namespace. When we have an annotation marking a service as global, the others are by default marked as local. Inversely in the other case, the others are marked as global. We can technically combine them into a single one but I wanted to separate out the annotations due to this reason

Reply:
I agree with Arthur's comment here. Having the export mode controlled by which annotation is used looks fragile and potentially confusing to me, compared to a straightforward configuration option.